news/_posts/2017-11-30-make-back-office-modules-great-again.md

```diff
@@ -0,0 +1,68 @@
+
+We have released a new theme system, introducing new helpers, widgets and inheritance system between templates: in the end it's now easier for everyone to make new themes with PrestaShop, specially for whom who are more designers than developers.
+
+We have started the migration of Back Office with the framework Symfony 2, first with Product and Module pages, then the Translation, Stock Management pages and we are actually migrating pages of Advanced Parameters section. We do it for so many reasons but to sum up, we want to focus as what we are the best: providing features for the e-merchants. Be able to Work on top of a professional, fully tested and documented framework is good for PrestaShop developers, Core team included.
+
+But it was not the main reason. The main reason is allowing every PrestaShop developer to use the Symfony ecosystem to make better, powerful and maintainable modules, to be able to evolve with the last features of PHP ecosystem.
+
+### Current situation of Back Office modules (pre 1.7.3)
+
+In Back Office, for the modern pages (migrated to Symfony 2), PrestaShop developers can still use Hooks like in the old ways. But they are not allowed to override templates, forms or data returned by the controllers.
+
+In Product page, a specific hook called `hookDisplayAdminProductsExtra` can display all modules into a specific tab and we have added some hooks at bottom, middle and top positions of each tab. However, it's not easy to hook at a specific position into the forms to add or re-order fields, buttons or blocks of information.
+
+Reviewing the actual hooks, they are not located wisely regarding the new UX of this page and some hooks are missing to make basic additions (like adding new buttons to menus for instance).
+
+![PrestaShop Hooks on Product Page](/assets/images/2017/11/product_page_hooks.png)
+
+Finally, and like in 1.6, it's still really difficult to know when, where and if a hook has been called, how many times and what are the modules hooked.
+
+### PrestaShop 1.7.3, the module experience re-invented
+
+The first new feature we'd like to present is the Hooks debugger. In all modern pages of Back Office, you have access to the PrestaShop debug toolbar, now enriched by a "Hooks" section.
+
+![PrestaShop Hooks Debugger section](/assets/images/2017/11/hooks_debugger_section.png)
+
+At first look, in every modern page you can know which and how many times hooks have been called. If you access the section panel, you have access to many additional information:
+
+* modules and hooks arguments
+* called and uncalled (but available) hooks
+* which modules react to hooks (the order is respected)
+
+![PrestaShop Hooks Debugger](/assets/images/2017/11/hooks_debugger.png)
+
+This allow every PrestaShop developer to choice wisely the right hook, but also to understand some behaviors if a specific hook doesn't render the expected view without have to enable/disable every module until find the guilty one.
```

Comment:
"It allows PrestaShop developers not only to pick the right hook but also to understand behaviors in case a specific hook would not fulfill its function (?). Now we can spot the guilty one without having to enable/disable each module."